modules/configuring-private-storage-endpoint-azure-vnet-subnet-iro-discovery.adoc

```diff
@@ -0,0 +1,154 @@
+
+.Prerequisites 
+
+* You have configured the image registry to run on Azure.
+* Your network has been set up using the Installer Provisioned Infrastructure (IPI) installation method. 
++
+For users with a custom network setup, see "Configuring a private storage endpoint on Azure with user-provided VNet and subnet names". 
+
+.Procedure 
+
+. Edit the Image Registry Operator `config` object and omit your VNet and subnet names:
++
+[source,terminal]
+----
+$ oc edit configs.imageregistry/cluster
+----
++
+[source,terminal]
+----
+# ...
+spec:
+  [...]
+   storage:
+      azure:
+        [...]
+        networkAccess:
+          type: Internal
+# ...
+----
+
+. Enter the following command to confirm that the Operator has completed provisioning. This might take a few minutes. 
++
+[source,terminal]
+----
+$ oc get configs.imageregistry/cluster -o=jsonpath="{.spec.storage.azure.privateEndpointName}" -w
+----
+
+. Optional. If the registry is exposed by a route, and you are configuring your storage account to be private, you must disable redirect if you want pulls external to the cluster to continue to work. Enter the following command to disable redirect on the Image Operator configuration:
++
+[source,terminal]
+----
+$ oc patch configs.imageregistry cluster --type=merge -p '{"spec":{"disableRedirect": true}}'
+----
++
+[NOTE]
+====
+When redirect is enabled,  pulling images from outside of the cluster will not work.
+====
+
+.Verification
+
+Use the following verification steps to ensure that users within the same cluster can pull images from the registry.
+
+. Fetch the registry service name by running the following command:
++
+[source,terminal]
+----
+$ oc registry info --internal=true
+----
++
+.Example output
++
+[source,terminal]
+----
+image-registry.openshift-image-registry.svc:5000
+----
+
+. Enter debug mode by running the following command:
++
+[source,terminal]
+----
+$ oc debug node/<node-name>
+----
+
+. Run the suggested `chroot` command. For example:
++
+[source,terminal]
+----
+$ chroot /host
+----
+
+. Enter the following command to log in to your container registry:
++
+[source,terminal]
+----
+$ podman login --tls-verify=false -u unused -p $(oc whoami -t) image-registry.openshift-image-registry.svc:5000
+----
++
+.Example output
++
+[source,terminal]
+----
+Login Succeeded!
+----
+
+. Enter the following command to verify that you can pull an image from the registry:
++
+[source,terminal]
+----
+$ podman pull --tls-verify=false image-registry.openshift-image-registry.svc:5000/openshift/tools
+----
++
+.Example output
++
+[source,terminal]
+----
+Trying to pull image-registry.openshift-image-registry.svc:5000/openshift/tools/openshift/tools...
+Getting image source signatures
+Copying blob 6b245f040973 done
+Copying config 22667f5368 done
+Writing manifest to image destination
+Storing signatures
+22667f53682a2920948d19c7133ab1c9c3f745805c14125859d20cede07f11f9
+----
+
+.Verification with redirect enabled
```

Comment:
Is this intended to run from within a node debug pod, like the previous verification step?

Reply:
Since redirect enabled will not work from outside the cluster and these steps are showing the failure they should be from outside the cluster..
With that said, the other procedure doesn't have this extra verification right? I think we can probably get rid of it.

Reply:
Removed.